src/helpers/github.ts

```diff
@@ -0,0 +1,101 @@
+import { retryAsync } from "ts-retry";
+import { Context } from "../types";
+
+export function parseGitHubUrl(url: string) {
+  const path = new URL(url).pathname.split("/");
+  if (path.length !== 5) {
+    throw new Error(`[parseGitHubUrl] Invalid url: [${url}]`);
+  }
+  return {
+    owner: path[1],
+    repo: path[2],
+    issue_number: Number(path[4]),
+  };
+}
+
+export type IssueParams = ReturnType<typeof parseGitHubUrl>;
+
+/**
+ * Gets the merge timeout depending on the status of the assignee. If there are multiple assignees with different
+ * statuses, the longest timeout is chosen.
+ */
+export async function getMergeTimeoutAndApprovalRequiredCount(context: Context, authorAssociation: string) {
+  const timeoutCollaborator = {
+    mergeTimeout: context.config.collaboratorMergeTimeout,
+    requiredApprovalCount: context.config.collaboratorMinimumApprovalsRequired,
+  };
+  const timeoutContributor = {
+    mergeTimeout: context.config.contributorMergeTimeout,
+    requiredApprovalCount: context.config.contributorMinimumApprovalsRequired,
+  };
+  return authorAssociation === "COLLABORATOR" || authorAssociation === "MEMBER" ? timeoutCollaborator : timeoutContributor;
```

Comment:
@gentlementlegen 

From https://michaelheap.com/github-actions-check-permission/:
"if the actor is a member of the organization that owns the repository their `author_association` will be `MEMBER`"

Shouldn't we also check for the `MEMBER` role here?

Reply:
Oh, my bad, the `MEMBER` check actually exists, so everything is fine here